## INTRODUCTION TO DATA ANALYSIS(PYTHON)
## PROJECT
### GEETANJALI SACHDEV (GROUP-B)


## IMPORTING LIBRARIES

In [191]:
!pip install bubbly
!pip install pandas-profiling

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [192]:
# for basic operations
import numpy as np
import pandas as pd
from pandas_profiling import ProfileReport


## IMPORTING DATA

### COMPANY DATAFRAME

In [193]:
# reading the company data
df_company = pd.read_excel('/content/2223MBD41_Project_Dataset-v01.xlsx',sheet_name='COMPANY')
df_company

COMPANY_NAME                                           CATEGORY  \
0            Twitter   Blogging Platforms, Messaging, SMS, Social Media   
1          Ant Group  Banking, E-Commerce, Financial Services, FinTe...   
2             Squire  Consumer, FinTech, Mobile Apps, Point of Sale,...   
3             Xiaomi             Consumer Electronics, Internet, Mobile   
4              maude  Consumer Goods, E-Commerce, Personal Health, W...   
..               ...                                                ...   
995            Valiu  Cryptocurrency, Finance, Financial Services, F...   
996          Airship  Analytics, Information Technology, Messaging, ...   
997          Jupiter                                Delivery, Logistics   
998  Bank of America  Asset Management, Banking, Finance, Financial ...   
999          Bitmain  Application Specific Integrated Circuit (ASIC)...   

                                     LOCATION    FOUNDED_ON     EXITED_ON  \
0    San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
1                   Hangzhou, Zhejiang, China      Oct 2014             —   
2           New York, New York, United States          2016             —   
3                     Beijing, Beijing, China   Apr 4, 2010   Jul 9, 2018   
4           Brooklyn, New York, United States          2018             —   
..                                        ...           ...           ...   
995       Bogotá, Distrito Especial, Colombia  Sep 17, 2018             —   
996           Portland, Oregon, United States   Jun 1, 2009             —   
997  San Francisco, California, United States          2019             —   
998  Charlotte, North Carolina, United States  Jul 31, 1998  Jan 13, 1978   
999                   Beijing, Beijing, China          2013             —   

    CLOSED_ON    REVENU_RANGE EMPLOYEE_NUMBER  
0           —     $1B to $10B       1001-5000  
1           —  $100M to $500M      5001-10000  
2           —     $1M to $10M          51-100  
3           —           $10B+          10001+  
4           —     $1M to $10M            1-10  
..        ...             ...             ...  
995         —     $1M to $10M           11-50  
996         —   $50M to $100M         251-500  
997         —   Less than $1M            1-10  
998         —           $10B+          10001+  
999         —     $1B to $10B         101-250  

[1000 rows x 8 columns]

In [194]:
# reading the investment data
df_investment = pd.read_excel('/content/2223MBD41_Project_Dataset-v01.xlsx',sheet_name='INVESTMENT')
df_investment

COMPANY_NAME              FUNDING_TYPE    MONEY_RAISED  \
0                   Pipe  Venture - Series Unknown    $250,000,000   
1                   Pipe                  Series A     $50,000,000   
2                   Pipe                      Seed      $6,000,000   
3                   Pipe                      Seed     $10,000,000   
4             Epic Games                      Seed  $1,000,000,000   
...                  ...                       ...             ...   
5932                 PTC           Post-IPO Equity  $1,000,000,000   
5933  Berkshire Hathaway           Post-IPO Equity      $3,510,000   
5934  Wipro Technologies           Post-IPO Equity               —   
5935  Wipro Technologies           Post-IPO Equity               —   
5936               Merck  Venture - Series Unknown      $5,000,000   

     ANNOUNCED_DATE     INVESTMENT_STAGE  
0      Mar 31, 2021                    —  
1       Mar 9, 2021  Early Stage Venture  
2      Feb 25, 2020                 Seed  
3      Jun 24, 2020                 Seed  
4      Apr 13, 2021                 Seed  
...             ...                  ...  
5932   Jun 11, 2018                    —  
5933   Aug 15, 2019                    —  
5934    Oct 1, 2006                    —  
5935    Sep 1, 2007                    —  
5936   Jan 21, 2016                    —  

[5937 rows x 5 columns]

In [195]:
# reading the acquisition data
df_acquisition = pd.read_excel('/content/2223MBD41_Project_Dataset-v01.xlsx',sheet_name='ACQUISITION')
df_acquisition

COMPANY_NAME       ACQUIREE_NAME ANNOUNCED_DATE           PRICE  \
0              IBM          Turbonomic   Apr 29, 2021  $2,000,000,000   
1              IBM           myInvenio   Apr 15, 2021               —   
2              IBM       Taos Mountain   Jan 14, 2021               —   
3              IBM            Cloudant   Feb 24, 2014               —   
4              IBM              TruQua   Nov 16, 2020               —   
...            ...                 ...            ...             ...   
5284      Opendoor       Open Listings   Sep 11, 2018               —   
5285     Inscripta  Solana Biosciences    Sep 5, 2018               —   
5286      Glossier              Dynamo   Feb 28, 2018               —   
5287  FreightWaves              StakUp    Jun 3, 2019               —   
5288         Auth0          Apility.io   Mar 12, 2020               —   

     ACQUISITION_TYPE  
0         Acquisition  
1         Acquisition  
2         Acquisition  
3         Acquisition  
4         Acquisition  
...               ...  
5284                —  
5285      Acquisition  
5286      Acquisition  
5287      Acquisition  
5288      Acquisition  

[5289 rows x 5 columns]

In [196]:
# reading the employee data
df_employee = pd.read_excel('/content/2223MBD41_Project_Dataset-v01.xlsx',sheet_name='EMPLOYEE')
df_employee

EMPLOYEE_MD5  \
0      aac4a9fe0edc50a77072eae64578a892   
1      f9c87bcba2ffe2237e039b2e2c86b16f   
2      36830ffebad9a630da8a7a03a45f2730   
3      84a91ad0f9b4149759f47fdc949fed42   
4      7881769fc62f54c36a687bda02b66198   
...                                 ...   
20697  72c32150cb3f9b6e806fa530ed610175   
20698  dd0ae519effadbef7559d299ad355f1f   
20699  aae97cfe2ea38ca0609a341e984b6b57   
20700  b8a8c88bce6444121ebce4ef8d41a204   
20701  c38442d6eaf4381a97379a05dc840f92   

                                    JOB_TITLES      COMPANY_NAME  \
0                                          CEO           Twitter   
1                                   Co-Founder           Twitter   
2                                  Team Member           Twitter   
3                              General Counsel           Twitter   
4                           VP, Trust & Safety           Twitter   
...                                        ...               ...   
20697                         Chairman and CEO         Iberdrola   
20698  Digital Channels Specialist Web Analyst         Iberdrola   
20699                                      CEO      JD Logistics   
20700                           Data Scientist  Keywords Studios   
20701                                      CEO              Citi   

                                        ATTENDED_SCHOOLS  
0                                                    NaN  
1       Northeastern University University of Massach...  
2                                                    NaN  
3       Cornell University New York University School...  
4                                                    NaN  
...                                                  ...  
20697                                                NaN  
20698                                EAE Business School  
20699                                                NaN  
20700                                                NaN  
20701                                                NaN  

[20702 rows x 4 columns]

## DATA PREPROCESSING

### COMPANY DATAFRAME

In [244]:
#covertimg columns into datetime format for easy readibility
df_company['FOUNDED_ON1'] = pd.to_datetime(df_company['FOUNDED_ON'])
df_company['EXITED_ON1'] = pd.to_datetime(df_company['EXITED_ON'])
df_company['CLOSED_ON1'] = pd.to_datetime(df_company['CLOSED_ON'])
df_company=df_company.replace("—","NaN") #replacing "—" with "NaN"
df_company

COMPANY_NAME                                           CATEGORY  \
0            Twitter   Blogging Platforms, Messaging, SMS, Social Media   
1          Ant Group  Banking, E-Commerce, Financial Services, FinTe...   
2             Squire  Consumer, FinTech, Mobile Apps, Point of Sale,...   
3             Xiaomi             Consumer Electronics, Internet, Mobile   
4              maude  Consumer Goods, E-Commerce, Personal Health, W...   
..               ...                                                ...   
995            Valiu  Cryptocurrency, Finance, Financial Services, F...   
996          Airship  Analytics, Information Technology, Messaging, ...   
997          Jupiter                                Delivery, Logistics   
998  Bank of America  Asset Management, Banking, Finance, Financial ...   
999          Bitmain  Application Specific Integrated Circuit (ASIC)...   

                                     LOCATION    FOUNDED_ON     EXITED_ON  \
0    San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
1                   Hangzhou, Zhejiang, China      Oct 2014           NaN   
2           New York, New York, United States          2016           NaN   
3                     Beijing, Beijing, China   Apr 4, 2010   Jul 9, 2018   
4           Brooklyn, New York, United States          2018           NaN   
..                                        ...           ...           ...   
995       Bogotá, Distrito Especial, Colombia  Sep 17, 2018           NaN   
996           Portland, Oregon, United States   Jun 1, 2009           NaN   
997  San Francisco, California, United States          2019           NaN   
998  Charlotte, North Carolina, United States  Jul 31, 1998  Jan 13, 1978   
999                   Beijing, Beijing, China          2013           NaN   

    CLOSED_ON   REVENUE_RANGE EMPLOYEE_NUMBER FOUNDED_ON1 EXITED_ON1  \
0         NaN     $1B to $10B       1001-5000  2006-03-21 2013-11-07   
1         NaN  $100M to $500M      5001-10000  2014-10-01        NaT   
2         NaN     $1M to $10M          51-100  2016-01-01        NaT   
3         NaN           $10B+          10001+  2010-04-04 2018-07-09   
4         NaN     $1M to $10M            1-10  2018-01-01        NaT   
..        ...             ...             ...         ...        ...   
995       NaN     $1M to $10M           11-50  2018-09-17        NaT   
996       NaN   $50M to $100M         251-500  2009-06-01        NaT   
997       NaN   Less than $1M            1-10  2019-01-01        NaT   
998       NaN           $10B+          10001+  1998-07-31 1978-01-13   
999       NaN     $1B to $10B         101-250  2013-01-01        NaT   

    CLOSED_ON1  
0          NaT  
1          NaT  
2          NaT  
3          NaT  
4          NaT  
..         ...  
995        NaT  
996        NaT  
997        NaT  
998        NaT  
999        NaT  

[1000 rows x 11 columns]

In [199]:
#renaming column REVENU_RANGE to REVENUE_RANGE
df_company=df_company.rename(columns={'REVENU_RANGE' : 'REVENUE_RANGE'})
df_company.head(5)

COMPANY_NAME                                           CATEGORY  \
0      Twitter   Blogging Platforms, Messaging, SMS, Social Media   
1    Ant Group  Banking, E-Commerce, Financial Services, FinTe...   
2       Squire  Consumer, FinTech, Mobile Apps, Point of Sale,...   
3       Xiaomi             Consumer Electronics, Internet, Mobile   
4        maude  Consumer Goods, E-Commerce, Personal Health, W...   

                                   LOCATION    FOUNDED_ON    EXITED_ON  \
0  San Francisco, California, United States  Mar 21, 2006  Nov 7, 2013   
1                 Hangzhou, Zhejiang, China      Oct 2014          NaN   
2         New York, New York, United States          2016          NaN   
3                   Beijing, Beijing, China   Apr 4, 2010  Jul 9, 2018   
4         Brooklyn, New York, United States          2018          NaN   

  CLOSED_ON   REVENUE_RANGE EMPLOYEE_NUMBER FOUNDED_ON1  
0       NaN     $1B to $10B       1001-5000  2006-03-21  
1       NaN  $100M to $500M      5001-10000  2014-10-01  
2       NaN     $1M to $10M          51-100  2016-01-01  
3       NaN           $10B+          10001+  2010-04-04  
4       NaN     $1M to $10M            1-10  2018-01-01

In [200]:
df_company.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   COMPANY_NAME     1000 non-null   object        
 1   CATEGORY         1000 non-null   object        
 2   LOCATION         1000 non-null   object        
 3   FOUNDED_ON       1000 non-null   object        
 4   EXITED_ON        1000 non-null   object        
 5   CLOSED_ON        1000 non-null   object        
 6   REVENUE_RANGE    1000 non-null   object        
 7   EMPLOYEE_NUMBER  1000 non-null   object        
 8   FOUNDED_ON1      1000 non-null   datetime64[ns]
dtypes: datetime64[ns](1), object(8)
memory usage: 70.4+ KB


In [201]:
# describing the data
df_company.describe(include='all').T

<ipython-input-201-b8786dd69808>:2: FutureWarning:

Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.



count unique  \
COMPANY_NAME     1000   1000   
CATEGORY         1000    977   
LOCATION         1000    255   
FOUNDED_ON       1000    428   
EXITED_ON        1000    269   
CLOSED_ON        1000      6   
REVENUE_RANGE    1000      9   
EMPLOYEE_NUMBER  1000     10   
FOUNDED_ON1      1000    370   

                                                               top freq  \
COMPANY_NAME                                               Twitter    1   
CATEGORY         Biotechnology, Health Care, Medical, Pharmaceu...    5   
LOCATION                  San Francisco, California, United States  173   
FOUNDED_ON                                                    2012   55   
EXITED_ON                                                      NaN  678   
CLOSED_ON                                                      NaN  995   
REVENUE_RANGE                                          $1M to $10M  230   
EMPLOYEE_NUMBER                                            101-250  196   
FOUNDED_ON1                                    2012-01-01 00:00:00   57   

                     first       last  
COMPANY_NAME           NaT        NaT  
CATEGORY               NaT        NaT  
LOCATION               NaT        NaT  
FOUNDED_ON             NaT        NaT  
EXITED_ON              NaT        NaT  
CLOSED_ON              NaT        NaT  
REVENUE_RANGE          NaT        NaT  
EMPLOYEE_NUMBER        NaT        NaT  
FOUNDED_ON1     1836-01-01 2020-05-01

In [202]:
# computer memory used by COMPANY dataframe in bytes
used_pc_memory = df_company.memory_usage(deep=True).sum()
used_pc_memory = used_pc_memory/(1000*1000)
print('Company data frame is using {0:.2f} MB computer memory'.format(used_pc_memory))

Company data frame is using 0.64 MB computer memory


### INVESTMENT DATAFRAME

In [203]:
#covertimg columns into datetime format for easy readibility
df_investment['ANNOUNCED_DATE1'] = pd.to_datetime(df_investment['ANNOUNCED_DATE'])
df_investment=df_investment.replace("—",'NaN') #replacing "—" with "NaN"
df_investment.head()

COMPANY_NAME              FUNDING_TYPE    MONEY_RAISED ANNOUNCED_DATE  \
0         Pipe  Venture - Series Unknown    $250,000,000   Mar 31, 2021   
1         Pipe                  Series A     $50,000,000    Mar 9, 2021   
2         Pipe                      Seed      $6,000,000   Feb 25, 2020   
3         Pipe                      Seed     $10,000,000   Jun 24, 2020   
4   Epic Games                      Seed  $1,000,000,000   Apr 13, 2021   

      INVESTMENT_STAGE ANNOUNCED_DATE1  
0                  NaN      2021-03-31  
1  Early Stage Venture      2021-03-09  
2                 Seed      2020-02-25  
3                 Seed      2020-06-24  
4                 Seed      2021-04-13

In [204]:
df_investment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5937 entries, 0 to 5936
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   COMPANY_NAME      5937 non-null   object        
 1   FUNDING_TYPE      5937 non-null   object        
 2   MONEY_RAISED      5937 non-null   object        
 3   ANNOUNCED_DATE    5937 non-null   object        
 4   INVESTMENT_STAGE  5937 non-null   object        
 5   ANNOUNCED_DATE1   5937 non-null   datetime64[ns]
dtypes: datetime64[ns](1), object(5)
memory usage: 278.4+ KB


In [205]:
# describing the data
df_investment.describe(include='all').T

<ipython-input-205-e2164806d227>:2: FutureWarning:

Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.



count unique                  top  freq      first       last
COMPANY_NAME      5937    935                Tesla    36        NaT        NaT
FUNDING_TYPE      5937     28             Series A   854        NaT        NaT
MONEY_RAISED      5937   1212                  NaN  1028        NaT        NaT
ANNOUNCED_DATE    5937   2331          Dec 8, 2020    25        NaT        NaT
INVESTMENT_STAGE  5937      5                  NaN  1766        NaT        NaT
ANNOUNCED_DATE1   5937   2331  2020-12-08 00:00:00    25 1968-07-31 2021-05-19

In [206]:
# computer memory used by INVESTMENT dataframe in bytes
used_pc_memory = df_investment.memory_usage(deep=True).sum()
used_pc_memory = used_pc_memory/(1000*1000)
print('Investment data frame is using {0:.2f} MB computer memory'.format(used_pc_memory))

Investment data frame is using 2.10 MB computer memory


### ACQUISITION DATAFRAME

In [207]:
#covertimg columns into datetime format for easy readibility
df_acquisition['ANNOUNCED_DATE1'] = pd.to_datetime(df_acquisition['ANNOUNCED_DATE'])
df_acquisition=df_acquisition.replace("—",'NaN') #replacing "—" with "NaN"
df_acquisition.head(10)

COMPANY_NAME          ACQUIREE_NAME ANNOUNCED_DATE           PRICE  \
0          IBM             Turbonomic   Apr 29, 2021  $2,000,000,000   
1          IBM              myInvenio   Apr 15, 2021             NaN   
2          IBM          Taos Mountain   Jan 14, 2021             NaN   
3          IBM               Cloudant   Feb 24, 2014             NaN   
4          IBM                 TruQua   Nov 16, 2020             NaN   
5          IBM              Nordcloud   Dec 21, 2020             NaN   
6          IBM                Instana   Nov 18, 2020             NaN   
7          IBM               7Summits   Jan 11, 2021             NaN   
8          IBM  Expertus Technologies   Dec 15, 2020             NaN   
9          IBM             NetObjects   Mar 19, 1997             NaN   

  ACQUISITION_TYPE ANNOUNCED_DATE1  
0      Acquisition      2021-04-29  
1      Acquisition      2021-04-15  
2      Acquisition      2021-01-14  
3      Acquisition      2014-02-24  
4      Acquisition      2020-11-16  
5      Acquisition      2020-12-21  
6      Acquisition      2020-11-18  
7      Acquisition      2021-01-11  
8      Acquisition      2020-12-15  
9      Acquisition      1997-03-19

In [208]:
df_acquisition.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5289 entries, 0 to 5288
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   COMPANY_NAME      5289 non-null   object        
 1   ACQUIREE_NAME     5289 non-null   object        
 2   ANNOUNCED_DATE    5289 non-null   object        
 3   PRICE             5289 non-null   object        
 4   ACQUISITION_TYPE  5289 non-null   object        
 5   ANNOUNCED_DATE1   5289 non-null   datetime64[ns]
dtypes: datetime64[ns](1), object(5)
memory usage: 248.0+ KB


In [209]:
# describing the data
df_acquisition.describe(include='all').T

<ipython-input-209-771a16726d48>:2: FutureWarning:

Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.



count unique                  top  freq      first       last
COMPANY_NAME      5289    497            Microsoft   235        NaT        NaT
ACQUIREE_NAME     5289   4167             Fastlane     4        NaT        NaT
ANNOUNCED_DATE    5289   2691          May 6, 2021    16        NaT        NaT
PRICE             5289    560                  NaN  3949        NaT        NaT
ACQUISITION_TYPE  5289      6          Acquisition  4087        NaT        NaT
ANNOUNCED_DATE1   5289   2659  2021-05-06 00:00:00    16 1923-01-01 2021-05-18

In [210]:
# computer memory used by ACQUISITION dataframe in bytes
used_pc_memory = df_acquisition.memory_usage(deep=True).sum()
used_pc_memory = used_pc_memory/(1000*1000)
print('Acquisition data frame is using {0:.2f} MB computer memory'.format(used_pc_memory))

Acquisition data frame is using 1.88 MB computer memory


### EMPLOYEE DATAFRAME

In [211]:
df_employee=df_employee.replace("—",'NaN') #replacing "—" with "NaN"
df_employee.tail(10)

EMPLOYEE_MD5  \
20692  13efc186592eba4ed93b0b61cc26330e   
20693  16e89da7ddb4cfa82ffcc99181abe3d4   
20694  33e456ce44957263e838c880b5aeda09   
20695  e5c660475f9caf75a69790515ae70758   
20696  0b8754de948e3648827a26827fda9df0   
20697  72c32150cb3f9b6e806fa530ed610175   
20698  dd0ae519effadbef7559d299ad355f1f   
20699  aae97cfe2ea38ca0609a341e984b6b57   
20700  b8a8c88bce6444121ebce4ef8d41a204   
20701  c38442d6eaf4381a97379a05dc840f92   

                                    JOB_TITLES                COMPANY_NAME  \
20692                                      CFO                         CAE   
20693    President and Chief Executive Officer                         CAE   
20694               Manager, Social Media Care  AMC Entertainment Holdings   
20695                                Film Crew  AMC Entertainment Holdings   
20696           Director - Smart Grid Planning                   Iberdrola   
20697                         Chairman and CEO                   Iberdrola   
20698  Digital Channels Specialist Web Analyst                   Iberdrola   
20699                                      CEO                JD Logistics   
20700                           Data Scientist            Keywords Studios   
20701                                      CEO                        Citi   

           ATTENDED_SCHOOLS  
20692          HEC Montréal  
20693                   NaN  
20694                   NaN  
20695                   NaN  
20696                   NaN  
20697                   NaN  
20698   EAE Business School  
20699                   NaN  
20700                   NaN  
20701                   NaN

In [212]:
df_employee.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20702 entries, 0 to 20701
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   EMPLOYEE_MD5      20702 non-null  object
 1   JOB_TITLES        20702 non-null  object
 2   COMPANY_NAME      20702 non-null  object
 3   ATTENDED_SCHOOLS  6584 non-null   object
dtypes: object(4)
memory usage: 647.1+ KB


In [213]:
# describing the data
df_employee.describe(include='all').T

count unique                               top  freq
EMPLOYEE_MD5      20702  20702  aac4a9fe0edc50a77072eae64578a892     1
JOB_TITLES        20702  12308                 Software Engineer   510
COMPANY_NAME      20702    964                         Microsoft  1756
ATTENDED_SCHOOLS   6584   2973               Stanford University   159

In [214]:
# computer memory used by EMPLOYEE dataframe in bytes
used_pc_memory = df_employee.memory_usage(deep=True).sum()
used_pc_memory = used_pc_memory/(1000*1000)
print('Employee data frame is using {0:.2f} MB computer memory'.format(used_pc_memory))

Employee data frame is using 6.01 MB computer memory


#### Removing Duplicates

In [215]:
# removing duplicate rows from all dataframe
df_company.drop_duplicates(inplace=True)
df_acquisition.drop_duplicates(inplace=True)
df_investment.drop_duplicates(inplace=True)
df_employee.drop_duplicates(inplace=True)

#### Data Profiling

In [216]:
#profile_company = pandas_profiling.ProfileReport(df_company)
#profile_company

In [217]:
#profile_investment = pandas_profiling.ProfileReport(df_investment)
#profile_investment

In [218]:
#profile_acquisition = pandas_profiling.ProfileReport(df_acquisition)
#profile_acquisition

In [219]:
#profile_employee = pandas_profiling.ProfileReport(df_employee)
#profile_employee

#### Cleaning MONEY_RAISED Column in INVESTEMENT DF.

In [220]:
df_investment['MONEY_RAISED1']= df_investment['MONEY_RAISED'].str.strip('$').str.strip('CA$').str.strip('₹').str.strip('€').str.strip('£').str.strip('A$').str.strip('¥').str.replace(',', '').str.replace('—', 'NaN').str.strip('N¥').str.strip('R$').str.replace('a','NaN').str.strip('ZAR').str.strip('SGD').str.strip('₩').str.strip('HF').str.strip('MYR').str.strip('EK').str.strip('UB')
df_investment['MONEY_RAISED1'].astype(float)
df_investment.head(20)

COMPANY_NAME              FUNDING_TYPE    MONEY_RAISED ANNOUNCED_DATE  \
0             Pipe  Venture - Series Unknown    $250,000,000   Mar 31, 2021   
1             Pipe                  Series A     $50,000,000    Mar 9, 2021   
2             Pipe                      Seed      $6,000,000   Feb 25, 2020   
3             Pipe                      Seed     $10,000,000   Jun 24, 2020   
4       Epic Games                      Seed  $1,000,000,000   Apr 13, 2021   
5       Epic Games  Venture - Series Unknown  $1,250,000,000   Oct 26, 2018   
6       Epic Games  Venture - Series Unknown  $1,530,000,000    Aug 6, 2020   
7       Epic Games  Venture - Series Unknown    $750,000,000    Jul 1, 2020   
8       Epic Games           Corporate Round    $250,000,000    Jul 9, 2020   
9       Epic Games          Secondary Market             NaN    May 8, 2020   
10            Brex                  Series D    $425,000,000   Apr 26, 2021   
11            Brex                  Series C    $100,000,000   Jun 11, 2019   
12            Brex                  Series C    $150,000,000   May 19, 2020   
13            Brex                  Series A      $7,000,000    Apr 1, 2017   
14            Brex            Debt Financing    $200,000,000   Dec 11, 2019   
15            Brex            Debt Financing    $100,000,000   Apr 16, 2019   
16            Brex          Secondary Market             NaN    Oct 1, 2019   
17            Brex                  Series C    $125,000,000    Oct 5, 2018   
20            Brex                  Series B     $50,000,000   Jun 19, 2018   
22  Animoca Brands  Venture - Series Unknown     $88,888,888   May 13, 2021   

       INVESTMENT_STAGE ANNOUNCED_DATE1 MONEY_RAISED1  
0                   NaN      2021-03-31     250000000  
1   Early Stage Venture      2021-03-09      50000000  
2                  Seed      2020-02-25       6000000  
3                  Seed      2020-06-24      10000000  
4                  Seed      2021-04-13    1000000000  
5                   NaN      2018-10-26    1250000000  
6                   NaN      2020-08-06    1530000000  
7                   NaN      2020-07-01     750000000  
8                   NaN      2020-07-09     250000000  
9                   NaN      2020-05-08           NaN  
10   Late Stage Venture      2021-04-26     425000000  
11   Late Stage Venture      2019-06-11     100000000  
12   Late Stage Venture      2020-05-19     150000000  
13  Early Stage Venture      2017-04-01       7000000  
14                  NaN      2019-12-11     200000000  
15                  NaN      2019-04-16     100000000  
16                  NaN      2019-10-01           NaN  
17   Late Stage Venture      2018-10-05     125000000  
20  Early Stage Venture      2018-06-19      50000000  
22                  NaN      2021-05-13      88888888

## DATA ANALYSIS

### How many companies have received at least one round of investment?

In [221]:
#How many companies have received at least one round of investment?
df_investment2=df_investment[(df_investment['INVESTMENT_STAGE']!='NaN')]
df_investment2['COMPANY_NAME'].count()

3351

Conclusion: 3351 COMPANIES have recieved at least one round of investment

###What are the distributions of the current status of these companies?

In [222]:
#the investment distributionof these companies
df_investment3=df_investment2.sort_values(['COMPANY_NAME'],ascending=False)
df_investment3[['COMPANY_NAME','INVESTMENT_STAGE']]
#a=df_investment3.groupby('COMPANY_NAME')[['INVESTMENT_STAGE']].value_counts()
#a

COMPANY_NAME     INVESTMENT_STAGE
2301  ultimate.ai                 Seed
2299  ultimate.ai  Early Stage Venture
2300  ultimate.ai                 Seed
2302  ultimate.ai                 Seed
2303  ultimate.ai                 Seed
...           ...                  ...
2055      23andMe   Late Stage Venture
1350          1mg  Early Stage Venture
1349          1mg  Early Stage Venture
1348          1mg  Early Stage Venture
1347          1mg   Late Stage Venture

[3351 rows x 2 columns]

###On average, how much money each company has received at Series A round?

In [223]:
#On average, how much money each company has received at Series A round?
cols_keep=['Series A']
df_investment4=df_investment[df_investment['FUNDING_TYPE'].isin(cols_keep)]
df_investment4['MONEY_RAISED1'].replace('NaN',int(0))
df_investment4.groupby(['COMPANY_NAME','FUNDING_TYPE'])['MONEY_RAISED1'].sum().to_frame().tail()


MONEY_RAISED1
COMPANY_NAME FUNDING_TYPE              
ipsy         Series A           2755795
monday.com   Series A           7600000
project44    Series A          10500000
tabby        Series A          23000000
ultimate.ai  Series A          20000000

###Which company has the largest number of acquisitions?

In [224]:
#Which company has the largest number of acquisitions?
df_new1=df_acquisition.groupby('COMPANY_NAME')['ACQUISITION_TYPE'].value_counts()
df3=df_new1.to_frame().add_suffix('_COUNT').reset_index()
df3=df3.loc[df3.ACQUISITION_TYPE=='Acquisition']
print(df3[df3.ACQUISITION_TYPE_COUNT == df3.ACQUISITION_TYPE_COUNT.max()])

  COMPANY_NAME ACQUISITION_TYPE  ACQUISITION_TYPE_COUNT
6    Accenture      Acquisition                     157


Conclusion : ACCENTURE have 157 number of Acquisitions

###Which of these companies have CEOs graduated from a top school?

In [225]:
#Which of these companies have CEOs graduated from a top school?
#df7 = df_employee[df_employee['JOB_TITLES'].isin(['CEO'])]
#df6=df_employee[df_employee['ATTENDED_SCHOOLS'].dropna()]
#df_employee['ATTENDED_SCHOOLS'].fillna('Others')
#d=df_employee[(df_employee['JOB_TITLES']=='CEO') & (df_employee['ATTENDED_SCHOOLS'] != 'Others')]
#d['ATTENDED_SCHOOLS'].dropna()

df=df_employee.dropna()
d=df[(df['JOB_TITLES']=='CEO')]
d[['JOB_TITLES','COMPANY_NAME','ATTENDED_SCHOOLS']]

JOB_TITLES         COMPANY_NAME  \
107          CEO               Google   
1879         CEO                Apple   
2470         CEO               Stripe   
2608         CEO  Amazon Web Services   
3625         CEO              Netflix   
...          ...                  ...   
20424        CEO              Corning   
20451        CEO           Plug Power   
20559        CEO                 QueQ   
20560        CEO            Exclaimer   
20621        CEO               ViaSat   

                                        ATTENDED_SCHOOLS  
107     Indian Institute of Technology Kharagpur Stan...  
1879                                     Duke University  
2470         Massachusetts Institute of Technology - MIT  
2608          Harvard Business School Harvard University  
3625                 Bowdoin College Stanford University  
...                                                  ...  
20424               Harvard University Lehigh University  
20451                                  Temple University  
20559   King Mongkut's Institute of Technology Ladkra...  
20560                            University of Greenwich  
20621                                    Rice University  

[69 rows x 3 columns]

Conclusion: 69 companies's CEO have graduated from a Top School

### How many Companies have closed?

In [226]:
#df_company['CLOSED_ON'].info()
(df_company['CLOSED_ON'] != "NaN").value_counts()

False    995
True       5
Name: CLOSED_ON, dtype: int64

Conclusion : 5 companies have closed so far.

### How amny Companies are located in France?

In [227]:
s=df_company['LOCATION']
F=df_company[s.str.contains('France')]
F['COMPANY_NAME'].count()


12

Conclusion : Out of all the companies , 12 comapnies are located in France

### How many Finance companies are located in United Stated?

In [228]:
FUS=df_company[(df_company['CATEGORY'].str.contains('Finance')) & (df_company['LOCATION'].str.contains('United States'))] 
FUS['COMPANY_NAME'].count()

54

Conclusion : Total 54 have located in United Stated that are working in Finance Sector

### How many companies are Founded before 01-01-2018?

In [229]:
FoundedBefore2018=(df_company['FOUNDED_ON1']<'2018-01-01').value_counts()
FoundedBefore2018

True     937
False     63
Name: FOUNDED_ON1, dtype: int64

Conclusion : 937 companies were founder before 2018 and 63 companies were founder after 2018

### Merging COMPANY and ACQUISITION Dataframe on COMPANY_NAME throght INNER JOIN

In [230]:
df_COMnACQ=df_company.merge(df_acquisition,on='COMPANY_NAME')
df_COMnACQ

COMPANY_NAME                                           CATEGORY  \
0             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
1             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
2             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
3             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
4             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
...               ...                                                ...   
4191  Bank of America  Asset Management, Banking, Finance, Financial ...   
4192  Bank of America  Asset Management, Banking, Finance, Financial ...   
4193  Bank of America  Asset Management, Banking, Finance, Financial ...   
4194  Bank of America  Asset Management, Banking, Finance, Financial ...   
4195          Bitmain  Application Specific Integrated Circuit (ASIC)...   

                                      LOCATION    FOUNDED_ON     EXITED_ON  \
0     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
1     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
2     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
3     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
4     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
...                                        ...           ...           ...   
4191  Charlotte, North Carolina, United States  Jul 31, 1998  Jan 13, 1978   
4192  Charlotte, North Carolina, United States  Jul 31, 1998  Jan 13, 1978   
4193  Charlotte, North Carolina, United States  Jul 31, 1998  Jan 13, 1978   
4194  Charlotte, North Carolina, United States  Jul 31, 1998  Jan 13, 1978   
4195                   Beijing, Beijing, China          2013           NaN   

     CLOSED_ON REVENUE_RANGE EMPLOYEE_NUMBER FOUNDED_ON1  \
0          NaN   $1B to $10B       1001-5000  2006-03-21   
1          NaN   $1B to $10B       1001-5000  2006-03-21   
2          NaN   $1B to $10B       1001-5000  2006-03-21   
3          NaN   $1B to $10B       1001-5000  2006-03-21   
4          NaN   $1B to $10B       1001-5000  2006-03-21   
...        ...           ...             ...         ...   
4191       NaN         $10B+          10001+  1998-07-31   
4192       NaN         $10B+          10001+  1998-07-31   
4193       NaN         $10B+          10001+  1998-07-31   
4194       NaN         $10B+          10001+  1998-07-31   
4195       NaN   $1B to $10B         101-250  2013-01-01   

                     ACQUIREE_NAME ANNOUNCED_DATE PRICE ACQUISITION_TYPE  \
0                           Scroll    May 4, 2021   NaN      Acquisition   
1                        Reshuffle   Mar 24, 2021   NaN        Acquihire   
2                          Breaker    Jan 4, 2021   NaN      Acquisition   
3                            Revue   Jan 26, 2021   NaN      Acquisition   
4                            UENO.    Jan 6, 2021   NaN      Acquisition   
...                            ...            ...   ...              ...   
4191                  LaSalle Bank    Oct 1, 2007   NaN      Acquisition   
4192   FleetBoston Financial Group   Oct 27, 2003   NaN      Acquisition   
4193         Countrywide Financial    Jul 1, 2008   NaN      Acquisition   
4194  TransMerit Merchant Services   Jul 13, 2004   NaN      Acquisition   
4195                    BlockTrail   Jul 20, 2016   NaN      Acquisition   

     ANNOUNCED_DATE1  
0         2021-05-04  
1         2021-03-24  
2         2021-01-04  
3         2021-01-26  
4         2021-01-06  
...              ...  
4191      2007-10-01  
4192      2003-10-27  
4193      2008-07-01  
4194      2004-07-13  
4195      2016-07-20  

[4196 rows x 14 columns]

#### How many Acquired Companies have more than 5000 Employees grouped by COMPANY_NAME?

In [231]:
df_COMnACQ['EMPLOYEE_NUMBER'].value_counts()

10001+        2319
1001-5000      686
501-1000       412
5001-10000     287
251-500        242
101-250        165
51-100          33
11-50           25
1-10            24
NaN              3
Name: EMPLOYEE_NUMBER, dtype: int64

In [232]:
g=df_COMnACQ[(df_COMnACQ['EMPLOYEE_NUMBER'].str.startswith('5001')) | (df_COMnACQ['EMPLOYEE_NUMBER'].str.contains('0001')) ]
g.groupby(['COMPANY_NAME','ACQUIREE_NAME'])['EMPLOYEE_NUMBER'].value_counts().to_frame()

EMPLOYEE_NUMBER
COMPANY_NAME ACQUIREE_NAME                     EMPLOYEE_NUMBER                 
Accenture    ?What If!                         10001+                         1
             AD.Dialeto Agencia de Publicidade 10001+                         1
             ASM Research, Inc.                10001+                         1
             ATAN                              10001+                         1
             Acceria                           10001+                         1
...                                                                         ...
Zomato       Runnr                             10001+                         1
             Sparse Labs                       10001+                         1
             TongueStun                        10001+                         1
             Urbanspoon                        10001+                         1
             gastronauci.pl                    10001+                         1

[2603 rows x 1 columns]

Conclusion : Total 2603 Acquired companies have more than 5000 employees

### Merging COMPANY and INVESTMENT Dataframe on COMPANY_NAME

In [233]:
df_COMnINV=df_company.merge(df_investment,on='COMPANY_NAME')
df_COMnINV

COMPANY_NAME                                           CATEGORY  \
0             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
1             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
2             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
3             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
4             Twitter   Blogging Platforms, Messaging, SMS, Social Media   
...               ...                                                ...   
4765  Bank of America  Asset Management, Banking, Finance, Financial ...   
4766          Bitmain  Application Specific Integrated Circuit (ASIC)...   
4767          Bitmain  Application Specific Integrated Circuit (ASIC)...   
4768          Bitmain  Application Specific Integrated Circuit (ASIC)...   
4769          Bitmain  Application Specific Integrated Circuit (ASIC)...   

                                      LOCATION    FOUNDED_ON     EXITED_ON  \
0     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
1     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
2     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
3     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
4     San Francisco, California, United States  Mar 21, 2006   Nov 7, 2013   
...                                        ...           ...           ...   
4765  Charlotte, North Carolina, United States  Jul 31, 1998  Jan 13, 1978   
4766                   Beijing, Beijing, China          2013           NaN   
4767                   Beijing, Beijing, China          2013           NaN   
4768                   Beijing, Beijing, China          2013           NaN   
4769                   Beijing, Beijing, China          2013           NaN   

     CLOSED_ON REVENUE_RANGE EMPLOYEE_NUMBER FOUNDED_ON1  \
0          NaN   $1B to $10B       1001-5000  2006-03-21   
1          NaN   $1B to $10B       1001-5000  2006-03-21   
2          NaN   $1B to $10B       1001-5000  2006-03-21   
3          NaN   $1B to $10B       1001-5000  2006-03-21   
4          NaN   $1B to $10B       1001-5000  2006-03-21   
...        ...           ...             ...         ...   
4765       NaN         $10B+          10001+  1998-07-31   
4766       NaN   $1B to $10B         101-250  2013-01-01   
4767       NaN   $1B to $10B         101-250  2013-01-01   
4768       NaN   $1B to $10B         101-250  2013-01-01   
4769       NaN   $1B to $10B         101-250  2013-01-01   

                  FUNDING_TYPE  MONEY_RAISED ANNOUNCED_DATE  \
0               Private Equity           NaN    Jan 5, 2020   
1     Venture - Series Unknown           NaN   May 14, 2010   
2             Secondary Market           NaN    Nov 1, 2012   
3                         Seed       $25,000    Jan 1, 2007   
4             Convertible Note           NaN    Mar 1, 2021   
...                        ...           ...            ...   
4765           Post-IPO Equity           NaN   Dec 17, 2018   
4766                  Series B  $422,000,000    Aug 7, 2018   
4767                  Series A   $50,000,000    Sep 4, 2017   
4768                  Series B  $292,700,000   Jun 11, 2018   
4769          Secondary Market           NaN    Aug 4, 2018   

         INVESTMENT_STAGE ANNOUNCED_DATE1 MONEY_RAISED1  
0          Private Equity      2020-01-05           NaN  
1                     NaN      2010-05-14           NaN  
2                     NaN      2012-11-01           NaN  
3                    Seed      2007-01-01         25000  
4                     NaN      2021-03-01           NaN  
...                   ...             ...           ...  
4765                  NaN      2018-12-17           NaN  
4766  Early Stage Venture      2018-08-07     422000000  
4767  Early Stage Venture      2017-09-04      50000000  
4768  Early Stage Venture      2018-06-11     292700000  
4769                  NaN      2018-08-04  

Grouping COMPANY_NAME and FUNDING_TYPE and displaying the count of INVESTMENT_STAGE 

In [234]:
df_COMnINV.groupby(['COMPANY_NAME','FUNDING_TYPE'])['INVESTMENT_STAGE'].value_counts().to_frame()

INVESTMENT_STAGE
COMPANY_NAME FUNDING_TYPE             INVESTMENT_STAGE                     
1mg          Debt Financing           NaN                                 2
             Series A                 Early Stage Venture                 1
             Series B                 Early Stage Venture                 2
             Series C                 Late Stage Venture                  1
             Venture - Series Unknown NaN                                 2
...                                                                     ...
tabby        Seed                     Seed                                1
             Series A                 Early Stage Venture                 1
ultimate.ai  Angel                    Seed                                1
             Seed                     Seed                                3
             Series A                 Early Stage Venture                 1

[3662 rows x 1 columns]

#### Sorting COMPANY_NAME in Ascending order and MONEY_RAISED in Descending Order and reseting Index

In [235]:
df_COMnINV=df_COMnINV.sort_values(['COMPANY_NAME','MONEY_RAISED'],ascending=[True,False]).reset_index()
df_COMnINV

index COMPANY_NAME                                           CATEGORY  \
0      3161          1mg  Delivery, E-Commerce, Health Care, Internet, M...   
1      3160          1mg  Delivery, E-Commerce, Health Care, Internet, M...   
2      3157          1mg  Delivery, E-Commerce, Health Care, Internet, M...   
3      3155          1mg  Delivery, E-Commerce, Health Care, Internet, M...   
4      3158          1mg  Delivery, E-Commerce, Health Care, Internet, M...   
...     ...          ...                                                ...   
4765   2800  ultimate.ai  Artificial Intelligence, CRM, Customer Service...   
4766   2798  ultimate.ai  Artificial Intelligence, CRM, Customer Service...   
4767   2799  ultimate.ai  Artificial Intelligence, CRM, Customer Service...   
4768   2796  ultimate.ai  Artificial Intelligence, CRM, Customer Service...   
4769   2797  ultimate.ai  Artificial Intelligence, CRM, Customer Service...   

                     LOCATION   FOUNDED_ON    EXITED_ON CLOSED_ON  \
0     Gurgaon, Haryana, India  Feb 3, 2015  Jul 5, 2016       NaN   
1     Gurgaon, Haryana, India  Feb 3, 2015  Jul 5, 2016       NaN   
2     Gurgaon, Haryana, India  Feb 3, 2015  Jul 5, 2016       NaN   
3     Gurgaon, Haryana, India  Feb 3, 2015  Jul 5, 2016       NaN   
4     Gurgaon, Haryana, India  Feb 3, 2015  Jul 5, 2016       NaN   
...                       ...          ...          ...       ...   
4765  Berlin, Berlin, Germany     Nov 2016          NaN       NaN   
4766  Berlin, Berlin, Germany     Nov 2016          NaN       NaN   
4767  Berlin, Berlin, Germany     Nov 2016          NaN       NaN   
4768  Berlin, Berlin, Germany     Nov 2016          NaN       NaN   
4769  Berlin, Berlin, Germany     Nov 2016          NaN       NaN   

      REVENUE_RANGE EMPLOYEE_NUMBER FOUNDED_ON1              FUNDING_TYPE  \
0      $10M to $50M           11-50  2015-02-03  Venture - Series Unknown   
1      $10M to $50M           11-50  2015-02-03  Venture - Series Unknown   
2      $10M to $50M           11-50  2015-02-03                  Series A   
3      $10M to $50M           11-50  2015-02-03            Debt Financing   
4      $10M to $50M           11-50  2015-02-03                  Series B   
...             ...             ...         ...                       ...   
4765  Less than $1M          51-100  2016-11-01                     Angel   
4766  Less than $1M          51-100  2016-11-01                      Seed   
4767  Less than $1M          51-100  2016-11-01                      Seed   
4768  Less than $1M          51-100  2016-11-01                  Series A   
4769  Less than $1M          51-100  2016-11-01                      Seed   

      MONEY_RAISED ANNOUNCED_DATE     INVESTMENT_STAGE ANNOUNCED_DATE1  \
0     ₹710,000,000   Jan 31, 2020                  NaN      2020-01-31   
1              NaN   May 31, 2016                  NaN      2016-05-31   
2       $6,000,000   Apr 20, 2015  Early Stage Venture      2015-04-20   
3      $17,800,000    Jul 3, 2020                  NaN      2020-07-03   
4      $16,000,000   Apr 20, 2016  Early Stage Venture      2016-04-20   
...            ...            ...                  ...             ...   
4765      €230,000    Mar 1, 2017                 Seed      2017-03-01   
4766           NaN   Sep 12, 2017                 Seed      2017-09-12   
4767    $3,500,000    Nov 4, 2019                 Seed      2019-11-04   
4768   $20,000,000    Dec 3, 2020  Early Stage Venture      2020-12-03   
4769    $1,300,000   Sep 18, 2018                 Seed      2018-09-18   

     MONEY_RAISED1  
0        710000000  
1              NaN  
2          6000000  
3         17800000  
4         16000000  
...            ...  
4765        230000  
4766           NaN  
4767       3500000  
4768      20000000  
4769       1300000  

[4770 rows x 16 columns]

#### No of companies with Revenue MORE than 10 billion and Funding type as Angel

In [236]:
j=df_COMnINV[(df_COMnINV['REVENUE_RANGE']=='$10B+') & (df_COMnINV['FUNDING_TYPE']=='Angel' )]
j['COMPANY_NAME'].count()


3

Conclusion : Total 3 Companies have Revenue Range of more tha 10 Billion and Funding type as Angel

### Merging INVESTMENT and ACQUISITION Dataframe on "COMPANY_NAME" BY LEFT JOIN

In [237]:
df_INVnACQ=df_investment.merge(df_acquisition,on='COMPANY_NAME',how='left')
df_INVnACQ

COMPANY_NAME              FUNDING_TYPE    MONEY_RAISED ANNOUNCED_DATE_x  \
0             Pipe  Venture - Series Unknown    $250,000,000     Mar 31, 2021   
1             Pipe                  Series A     $50,000,000      Mar 9, 2021   
2             Pipe                      Seed      $6,000,000     Feb 25, 2020   
3             Pipe                      Seed     $10,000,000     Jun 24, 2020   
4       Epic Games                      Seed  $1,000,000,000     Apr 13, 2021   
...            ...                       ...             ...              ...   
17620        Merck  Venture - Series Unknown      $5,000,000     Jan 21, 2016   
17621        Merck  Venture - Series Unknown      $5,000,000     Jan 21, 2016   
17622        Merck  Venture - Series Unknown      $5,000,000     Jan 21, 2016   
17623        Merck  Venture - Series Unknown      $5,000,000     Jan 21, 2016   
17624        Merck  Venture - Series Unknown      $5,000,000     Jan 21, 2016   

          INVESTMENT_STAGE ANNOUNCED_DATE1_x MONEY_RAISED1  \
0                      NaN        2021-03-31     250000000   
1      Early Stage Venture        2021-03-09      50000000   
2                     Seed        2020-02-25       6000000   
3                     Seed        2020-06-24      10000000   
4                     Seed        2021-04-13    1000000000   
...                    ...               ...           ...   
17620                  NaN        2016-01-21       5000000   
17621                  NaN        2016-01-21       5000000   
17622                  NaN        2016-01-21       5000000   
17623                  NaN        2016-01-21       5000000   
17624                  NaN        2016-01-21       5000000   

                              ACQUIREE_NAME ANNOUNCED_DATE_y           PRICE  \
0                                       NaN              NaN             NaN   
1                                       NaN              NaN             NaN   
2                                       NaN              NaN             NaN   
3                                       NaN              NaN             NaN   
4                                ArtStation     Apr 30, 2021             NaN   
...                                     ...              ...             ...   
17620                    Sirna Therapeutics     Oct 30, 2006  $1,100,000,000   
17621                 Calporta Therapeutics     Nov 13, 2019    $576,000,000   
17622  Suzhou Taizhu Technology Development      Sep 9, 2009             NaN   
17623                                ArQule      Dec 9, 2019  $2,700,000,000   
17624               Inspire Pharmaceuticals      Apr 5, 2011    $430,000,000   

      ACQUISITION_TYPE ANNOUNCED_DATE1_y  
0                  NaN               NaT  
1                  NaN               NaT  
2                  NaN               NaT  
3                  NaN               NaT  
4          Acquisition        2021-04-30  
...                ...               ...  
17620      Acquisition        2006-10-30  
17621      Acquisition        2019-11-13  
17622      Acquisition        2009-09-09  
17623      Acquisition        2019-12-09  
17624      Acquisition        2011-04-05  

[17625 rows x 12 columns]

## DATA VISUALIZATION

In [238]:
df_employee['ATTENDED_SCHOOLS'].value_counts().plot.hist(bins=15)

In [239]:
df_acquisition['ACQUISITION_TYPE'].value_counts().plot.barh()

In [240]:
df_investment['FUNDING_TYPE'].value_counts().plot.bar()

In [241]:
df_company[['EMPLOYEE_NUMBER','REVENUE_RANGE']].value_counts().plot.box()